opencti-platform/opencti-graphql/src/schema/schema-attributes.ts

```diff
@@ -218,3 +217,82 @@ export const isObjectFlatAttribute = (k: string): boolean => {
 export const isMultipleAttribute = (entityType: string, k: string): boolean => (
   k.startsWith(RULE_PREFIX) || schemaAttributesDefinition.isMultipleAttribute(entityType, k)
 );
+
+// -- utility functions independent of attribute registration --
+// (inner mappings are not registered like first-level attribute)
+
+export const isMandatoryAttributeMapping = (schemaDef: AttributeDefinition) => schemaDef.mandatoryType === 'external' || schemaDef.mandatoryType === 'internal';
+
+export const isNonFLatObjectAttributeMapping = (schemaDef: AttributeDefinition) : schemaDef is (ObjectAttribute | NestedObjectAttribute) => { // handy typeguard
+  return schemaDef.type === 'object' && schemaDef.format !== 'flat';
+};
+
+/**
+ * Validates that the given input conforms to the constraints in the corresponding schema definition.
+ * Recursively checks non-flat objects mappings.
+ * @param input an input object candidate to indexing in elastic
+ * @param schemaDef AttributeDefinition for the given input data
+ */
+const validateInputAgainstSchema = (input: any, schemaDef: AttributeDefinition) => {
+  const isMandatory = isMandatoryAttributeMapping(schemaDef);
+  if (isMandatory && R.isNil(input)) {
+    throw FunctionalError(`Validation against schema failed on attribute [${schemaDef.name}]: this mandatory field cannot be nil`, { value: input });
+  }
+
+  if (isNonFLatObjectAttributeMapping(schemaDef)) {
+    // check 'multiple' constraint
+    if (isMandatory) {
+      if (schemaDef.multiple && !Array.isArray(input)) {
+        throw FunctionalError(`Validation against schema failed on attribute [${schemaDef.name}]: value must be an array`, { value: input });
+      }
+      if (!schemaDef.multiple && (Array.isArray(input) || !R.is(Object, input))) {
+        throw FunctionalError(`Validation against schema failed on attribute [${schemaDef.name}]: value must be an object`, { value: input });
+      }
+    }
+    if (!isMandatory && R.isNil(input)) {
+      return; // nothing to check (happens on 'remove' operation for instance
+    }
+
+    const inputValues = Array.isArray(input) ? input : [input];
+    inputValues.forEach((value) => {
+      // check the value adhere to its mapping
+      const valueKeys = Object.keys(value);
+      schemaDef.mappings.forEach((mapping) => {
+        // mandatory fields: the value must have a field with this name
+        if (isMandatoryAttributeMapping(mapping) && !valueKeys.includes(mapping.name)) {
```

Comment:
what if the input is like: `{ myKey: null }`, `myKey` is include in the array of keys but the value is still invalid no?

Reply:
ok maybe this check is done in the recursive step below

Reply:
yes, in recursion it's caught